onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -196,6 +233,18 @@
   return nullptr;
 }
 
+bool isHardwareAdapter(ComPtr<IDXCoreAdapter> adapter) {
+    bool isHardware{ false };
+    THROW_IF_FAILED(adapter->GetProperty(
+        DXCoreAdapterProperty::IsHardware,
+        &isHardware));
+    return isHardware;
+}
+
+bool supportsGraphics(ComPtr<IDXCoreAdapter> adapter) {
```

Comment:
>supportsGraphics [](http://example.com/codeflow?start=5&length=16)

Changing semantics of this method to be "IsGPU" and adding another "IsNPU" can make the logic easier to understand here.